feat: Add ascending sort function

LukasRiver opened the pull request:
Added a function to sort numbers and strings in ascending order.

LukasRiver committed:
feat: Add ascending sort function

Added a function to sort numbers and strings in ascending order.

hcw0915 reviewed (commented):
後續可以考慮 把 升冪 降冪  做在同一支 
- 應該也不用特別選擇 str/number 直接裡面判斷就好
- 然後如果 是  `[ { } , { } ,{ } ]` 這種形式的話 可以考慮要排序的 key 給使用者選擇要哪一個。
```
[
 {id: 0, fruit: 'banana'},
 {id: 1, fruit: 'apple'},
 {id: 2, fruit: 'cheery'},
]
```

所以個人最理想是這樣 fn(array, "id" , "dec" )
看是要用 boolean 表達升降 還是要另外用 str 表示。

Antonio-Frontend reviewed (commented):
okok

Commit:
Merge remote-tracking branch 'origin/main' into James

LukasRiver committed:
refactor: Enhance sortArray function for clarity

- Validate input to ensure array type

- Check all objects in array to have the specified key before sorting

- Rename variable names for clarity: firstValue -> valueA, secondValue -> valueB

- Handle edge cases for string and number sorting

- Update JSDoc comments for better documentation

hcw0915 reviewed (approved):
OKOK 辛苦辛苦，
我覺得因為我們後續會考慮到各種變數狀況，
所以可能可以討論一個問題就是檢測型別過不過   
=>   checkType(data, 'type') 這樣
然後後續 refactor 函數讓他變簡潔~
